### PR TITLE
Add replication factor metrics and add database tag in throughput metrics and schema metrics

### PR DESCRIPTION
**Pull request by SpriCoder**
Add replication factor metrics and add database tag in throughput metrics and schema metrics

### CODE REVIEW

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 1]
```diff
@@ -167,6 +172,25 @@ public TSStatus setDatabase(DatabaseSchemaPlan databaseSchemaPlan) {
           MetricService.getInstance(), configManager, databaseSchemaPlan.getSchema().getName());
       // Adjust the maximum RegionGroup number of each Database
       adjustMaxRegionGroupNum();
+      // Add database replication factor metrics
+      metricService
+          .getOrCreateGauge(
+              Metric.REPLICATION_FACTOR.toString(),
+              MetricLevel.CORE,
+              Tag.TYPE.toString(),
+              "data",
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:181]: do not introduce magic string?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 2]
```diff
@@ -167,6 +172,25 @@ public TSStatus setDatabase(DatabaseSchemaPlan databaseSchemaPlan) {
...
+              Tag.DATABASE.toString(),
+              databaseSchemaPlan.getSchema().getName())
+          .set(databaseSchemaPlan.getSchema().dataReplicationFactor);
+      metricService
+          .getOrCreateGauge(
+              Metric.REPLICATION_FACTOR.toString(),
+              MetricLevel.CORE,
+              Tag.TYPE.toString(),
+              "schema",
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:190]: same

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 3]
```diff
@@ -248,6 +292,21 @@ public TSStatus deleteDatabase(DeleteDatabasePlan deleteDatabasePlan) {
     TSStatus result;
     try {
       result = getConsensusManager().write(deleteDatabasePlan);
+      // Remove database replication factor metric
+      metricService.remove(
+          MetricType.GAUGE,
+          Metric.REPLICATION_FACTOR.toString(),
+          Tag.TYPE.toString(),
+          "data",
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:300]: same

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 4]
```diff
@@ -248,6 +292,21 @@ public TSStatus deleteDatabase(DeleteDatabasePlan deleteDatabasePlan) {
...
+          Tag.DATABASE.toString(),
+          deleteDatabasePlan.getName());
+      metricService.remove(
+          MetricType.GAUGE,
+          Metric.REPLICATION_FACTOR.toString(),
+          Tag.TYPE.toString(),
+          "schema",
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:307]: same

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 5]
```diff
@@ -234,7 +258,27 @@ public TSStatus alterDatabase(DatabaseSchemaPlan databaseSchemaPlan) {
...
+              "data",
+              Tag.DATABASE.toString(),
+              databaseSchemaPlan.getSchema().getName())
+          .set(databaseSchemaPlan.getSchema().dataReplicationFactor);
+      metricService
+          .getOrCreateGauge(
+              Metric.REPLICATION_FACTOR.toString(),
+              MetricLevel.CORE,
+              Tag.TYPE.toString(),
+              "schema",
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:277]: same

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java [hunk 6]
```diff
@@ -234,7 +258,27 @@ public TSStatus alterDatabase(DatabaseSchemaPlan databaseSchemaPlan) {
...
     try {
-      return getConsensusManager().write(databaseSchemaPlan);
+      result = getConsensusManager().write(databaseSchemaPlan);
+      // Alter database replication factor metrics
+      metricService
+          .getOrCreateGauge(
+              Metric.REPLICATION_FACTOR.toString(),
+              MetricLevel.CORE,
+              Tag.TYPE.toString(),
+              "data",
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:268]: same

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/memtable/AbstractMemTable.java [hunk 7]
```diff
@@ -609,6 +633,12 @@ public void deserialize(DataInputStream stream) throws IOException {
     totalPointsNumThreshold = stream.readLong();
     maxPlanIndex = stream.readLong();
     minPlanIndex = stream.readLong();
+    int databaseLength = stream.readInt();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/memtable/AbstractMemTable.java:636]: This seems likely to cause compatibility issues. Can we restore the database by some other means at recovery time without serializing it

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/metric/SchemaEngineMemMetric.java [hunk 8]
```diff
@@ -58,14 +58,18 @@ public void bindTo(AbstractMetricService metricService) {
         engineStatistics,
         ISchemaEngineStatistics::getTotalSeriesNumber,
         Tag.NAME.toString(),
-        TIME_SERES_CNT);
+        TIME_SERES_CNT,
+        Tag.DATABASE.toString(),
+        "total");
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/metric/SchemaEngineMemMetric.java:63]: So do we still need this field statistic? Instead, use grafana to calculate the value of all databases in the current node using sum

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Awesome work！
PTAL

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java (hunk 1 above)
Comment (SpriCoder) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/schema/ClusterSchemaManager.java:181]: Ok

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/metric/SchemaEngineMemMetric.java (hunk 8 above)
Comment (SpriCoder) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/metric/SchemaEngineMemMetric.java:63]: Remove and we can use SCHEMA_REGION metric to statistic

File: iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/recover/file/TsFilePlanRedoerTest.java [hunk 9]
```diff
@@ -69,6 +69,7 @@
 
 public class TsFilePlanRedoerTest {
   private static final String SG_NAME = "root.recover_sg";
+  private static final String DATA_REGION_ID = "DataRegion-1";
```
Comment (HTHou) [inline iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/recover/file/TsFilePlanRedoerTest.java:72]: Should be a digital number?

File: iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/recover/WALRecoverManagerTest.java [hunk 10]
```diff
@@ -88,6 +88,7 @@ public class WALRecoverManagerTest {
   private static final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
   private static final CommonConfig commonConfig = CommonDescriptor.getInstance().getConfig();
   private static final String SG_NAME = "root.recover_sg";
+  private static final String DATA_REGION_ID = "DataRegion-1";
```
Comment (HTHou) [inline iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/wal/recover/WALRecoverManagerTest.java:91]: Same as above.

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**